### PR TITLE
Add functions to load CSV in Google Cloud Storage into a BQ table

### LINKED ISSUE #126: [FR] Function to load a table from Google Cloud Storage object(s)?
BQ allows [loading data directly from cloud storage](https://cloud.google.com/bigquery/docs/loading-data-cloud-storage). Currently `bigrquery` has a `insert_upload_job` function that writes a temp csv file from a data frame and loads the temp csv into a BQ table. Can we also have a function to load a table from one or many objects in Cloud Storage?  I think it can be done by using a simple `POST` to `https://www.googleapis.com/bigquery/v2/` with a `configuration.load` payload specifying the meta information. One extra bit we need to write to enable this might be to expand the scope of `get_access_cred` to allow accessing cloud storage.

Is someone already working on this feature? If not, I can take a stab and send out a pull request.

### PR DESCRIPTION
**Pull request by jarodmeng**
Fix #126 

`insert_load_csv_from_cs_job()` inserts an [insert.load](https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load) job to load a CSV file stored in Cloud Storage into BQ as a table. `load_csv_from_cs()` is a wrapper that waits for the insert job to finish.

### CODE REVIEW

**Comment by iainmwallace** [comment]
This is great feature - any chance that we could resolve the NEWS.md file conflict?

**Comment by everron** [comment]
This PR would be of great help. I am upping this. Plus, the NEWS.md conflict seems very minor

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Version 0.3.0.9000
 
+* Add `insert_load_csv_from_cs_job()` and `load_csv_from_cs()` functions to help
```
Comment (hadley) [inline NEWS.md]: Can you please move up to the current version?

File: R/cs.r [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+#' This is a low-level function that creates an insert job. To wait until it is
+#' finished and retrive the new table, see [load_csv_from_cs()]
+#'
+#' @param project The project name, a string
+#' @param dataset The name of the dataset that contains the destination table,
+#'   a string
+#' @param table The name of the table to load data into
+#' @param source_uris The fully-qualified URIs that point to a CSV file in
+#'   Google Cloud Storage; see
+#'   \href{https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.load.sourceUris}{the API documentation}
```
Comment (hadley) [inline R/cs.r]: Can you please use markdown syntax here?

File: R/cs.r [hunk 3]
```diff
@@ -0,0 +1,93 @@
+#' Insert a job to load a CSV file from Google Cloud Storage into a table
+#'
+#' This is a low-level function that creates an insert job. To wait until it is
```
Comment (hadley) [inline R/cs.r]: Maybe this should be documented in the same file as `load_csv_from_cs()`?

File: R/cs.r [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                                        write_disposition = "WRITE_EMPTY",
+                                        ...) {
+  assert_that(is.string(project), is.string(dataset), is.string(table))
+
+  url <- bq_path(project, jobs = "")
+
+  body <- list(
+    configuration = list(
+      load = list(
+        sourceFormat = "CSV",
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/cs.r]: Is there an obvious reason this shouldn't be an argument?

File: R/cs.r [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+#'   \code{"WRITE_TRUNCATE"} and \code{"WRITE_APPEND"}; see
+#'   \href{https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.load.writeDisposition}{the API documentation}
+#'   for more information
+#' @family jobs
+#' @return a job resource list, as documented at
+#'   \url{https://developers.google.com/bigquery/docs/reference/v2/jobs}
+#' @seealso API documentation for insert method:
+#'   \url{https://developers.google.com/bigquery/docs/reference/v2/jobs/insert}
+#' @export
+insert_load_csv_from_cs_job <- function(project, dataset, table,
```
Comment (hadley) [inline R/cs.r]: I think this could just be `insert_load_job()`

**Comment by hadley** [comment]
Done in bcce9b72f06a11f8d3e12687262db445f28863bc

**Pull request closed without merging**